loader/loader.c

```diff
@@ -1492,9 +1492,15 @@ struct loader_icd_term *loader_icd_add(struct loader_instance *ptr_inst, const s
     icd_term->scanned_icd = scanned_icd;
     icd_term->this_instance = ptr_inst;
 
-    // Prepend to the list
-    icd_term->next = ptr_inst->icd_terms;
-    ptr_inst->icd_terms = icd_term;
+    // Append to the list
+    struct loader_icd_term *prev = ptr_inst->icd_terms;
+    if (prev == NULL) {
+        ptr_inst->icd_terms = icd_term;
+    } else {
+        while (prev->next)
+            prev = prev->next;
+        prev->next = icd_term;
+    }
```

Comment:
It would be nice to track the list tail to avoid iterating the entire list here, but probably a moot point with realistic numbers of ICDs, so not a big deal. This way is certainly less complex.